### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by hellwolf**
- Dependency updates:
  - update all dependencies, targeting minor versions
  - update major versions of:
    - hardhat 9
    - syncpack
    - solhint
    - webpack-cli 5
    - prettier-eslint, @typescript-eslint/eslint-plugin, @typescript-eslint/parser
    - @nomicfoundation/hardhat-ethers
    - node-jq
    - chai-as-promised
    - nyc
    - @graphql-codegen/near-operation-file-preset
- syncpack linting:
  - integrated into ci and pre-commit
  - format all package.json
  - pin stack-trace package version due to its es module usage
- removed dependency resolutions:
  - root: mocha
  - sdk-core: graphql
- upgrade nodejs usage
  - making node22 default
  - testing matrix expanded to 18,20,22
- update to yarn 1.22.22

**Dependencies left not updated to new major versions**

◉   @nomicfoundation/hardhat-chai-matchers 1.0.6 → 2.0.7  - https://github.com/nomicfoundation/hardhat
◉   eslint 8.57.0 → 9.9.1                          - https://github.com/eslint/eslint
◉   web3 ^1.10.4 → 4.12.1                           - https://github.com/ChainSafe/web3.js
◉   @openzeppelin/contracts 4.9.6 → 5.0.2         - https://github.com/OpenZeppelin/openzeppelin-contracts
◉   ethers ^5.7.2 → 6.13.2                         - https://github.com/ethers-io/ethers.js
◉   auto-bind 4.0.0 → 5.0.1                       - https://github.com/sindresorhus/auto-bind
◉   node-fetch 2.7.0 → 3.3.2                      - https://github.com/node-fetch/node-fetch
◉   graphql-request ^6.1.0 → 7.1.0                 - https://github.com/jasonkuhrt/graphql-request
◉   @reduxjs/toolkit ^1.9.7 → 2.2.7                - https://github.com/reduxjs/redux-toolkit
◉   react-redux ^8.1.3 → 9.1.2                     - https://github.com/reduxjs/react-redux

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -2,6 +2,7 @@
 . "$(dirname "$0")/_/husky.sh"
 
 # always lint these
+npm run lint:syncpack
```
Comment (hellwolf) [inline .husky/pre-commit:14]: might be assuming too much: if you don't use nix, you need to have shellcheck and actionlint installed globally.

Comment (kasparkallas) [reply .husky/pre-commit:14]: `shellcheck` can be installed from NPM and could be added as a dev dependency to the project. Doesn't seem to be the case for `actionlint`.

File: package.json [hunk 2]
```diff
@@ -1,16 +1,58 @@
...
+        "@nomiclabs/hardhat-etherscan": "^3.1.7",
+        "@nomiclabs/hardhat-web3": "^2.0.0",
+        "@openzeppelin/test-helpers": "^0.5.16",
+        "@truffle/hdwallet-provider": "^2.1.15",
+        "@typechain/ethers-v5": "^11.1.2",
+        "@typechain/hardhat": "9.1.0",
+        "@types/lodash": "^4.17.7",
+        "@types/mocha": "^10.0.7",
+        "@typescript-eslint/eslint-plugin": "8.3.0",
+        "@typescript-eslint/parser": "8.3.0",
```
Comment (kasparkallas) [inline package.json]: Not sure if some of these locked versions are on purpose

**Review by kasparkallas — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (hellwolf) [reply package.json]: probably not, not sure why it happenned with syncpack updating them

File: .husky/pre-commit (hunk 1 above)
Comment (hellwolf) [reply .husky/pre-commit:14]: let me clean up a bit of the shell scripts that have these dependencies.

Comment (hellwolf) [reply .husky/pre-commit:14]: Cleaned up with some warning messages.

File: package.json (hunk 2 above)
Comment (hellwolf) [reply package.json]: Codified the rules with the syncpack:

```
semverGroups:
  # Note: rule priority is top (high) to bottom (low)
  - label: Use looser ranges for metadata and ethereum-contracts local packages
    dependencies:
      - "@superfluid-finance/metadata"
      - "@superfluid-finance/ethereum-contracts"
    range: "^"
  - label: Use exact version for prod dependencies
    dependencyTypes: ["prod"]
    range: ""
  - label: Use looser range by default
    range: "^"

```

**Comment by d10r** [comment]
getting this when committing in ethereum-contracts:
```
husky - DEPRECATED

Please remove the following two lines from .husky/pre-commit:

#!/usr/bin/env sh
. "$(dirname -- "$0")/_/husky.sh"

They WILL FAIL in v10.0.0
```

**Comment by d10r** [comment]
new behaviour:
```
Versions
= Pin stack-trace to the same version, since newer version starts to use ES module 
     1 ✓ already valid
= Default Version Group ========================================================
✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/ethereum-contracts/package.json > devDependencies [LocalPackageMismatch]
✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/js-sdk/package.json > dependencies [LocalPackageMismatch]
✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/sdk-core/package.json > dependencies [LocalPackageMismatch]
✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/subgraph/package.json > devDependencies [LocalPackageMismatch]
✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/automation-contracts/scheduler/package.json > devDependencies [LocalPackageMismatch]
✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/automation-contracts/autowrap/package.json > devDependencies [LocalPackageMismatch]
   117 ✓ already valid
     6 ✓ can be auto-fixed
Semver Ranges
= the version property of package.json files must always be exact ==============
✓ 11 valid
= Use looser ranges for metadata and ethereum-contracts local packages =========
✓ 11 valid
= Use exact version for prod dependencies ======================================
✓ 23 valid
= Use looser range by default ==================================================
✓ 79 valid
husky - pre-commit script failed (code 1)
```

I used `yarn manage-versions` to fix, which had this side effect:
```
+++ b/packages/subgraph/package.json
@@ -5,11 +5,11 @@
     "dependencies": {
         "@graphprotocol/graph-cli": "0.80.1",
         "@graphprotocol/graph-ts": "0.35.1",
-        "@superfluid-finance/sdk-core": "0.8.0",
+        "@superfluid-finance/sdk-core": "^0.8.0",
```

probably not intended (?)

**Comment by hellwolf** [comment]
> getting this when committing in ethereum-contracts:
> 
> ```
> husky - DEPRECATED
> 
> Please remove the following two lines from .husky/pre-commit:
> 
> #!/usr/bin/env sh
> . "$(dirname -- "$0")/_/husky.sh"
> 
> They WILL FAIL in v10.0.0
> ```

fixed

**Review by d10r — APPROVED** [review]

**Comment by hellwolf** [comment]
> new behaviour:
> 
> ```
> Versions
> = Pin stack-trace to the same version, since newer version starts to use ES module 
>      1 ✓ already valid
> = Default Version Group ========================================================
> ✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/ethereum-contracts/package.json > devDependencies [LocalPackageMismatch]
> ✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/js-sdk/package.json > dependencies [LocalPackageMismatch]
> ✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/sdk-core/package.json > dependencies [LocalPackageMismatch]
> ✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/subgraph/package.json > devDependencies [LocalPackageMismatch]
> ✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/automation-contracts/scheduler/package.json > devDependencies [LocalPackageMismatch]
> ✘ @superfluid-finance/metadata ^1.4.1 → ^1.5.0 packages/automation-contracts/autowrap/package.json > devDependencies [LocalPackageMismatch]
>    117 ✓ already valid
>      6 ✓ can be auto-fixed
> Semver Ranges
> = the version property of package.json files must always be exact ==============
> ✓ 11 valid
> = Use looser ranges for metadata and ethereum-contracts local packages =========
> ✓ 11 valid
> = Use exact version for prod dependencies ======================================
> ✓ 23 valid
> = Use looser range by default ==================================================
> ✓ 79 valid
> husky - pre-commit script failed (code 1)
> ```
> 
> I used `yarn manage-versions` to fix, which had this side effect:
> 
> ```
> +++ b/packages/subgraph/package.json
> @@ -5,11 +5,11 @@
>      "dependencies": {
>          "@graphprotocol/graph-cli": "0.80.1",
>          "@graphprotocol/graph-ts": "0.35.1",
> -        "@superfluid-finance/sdk-core": "0.8.0",
> +        "@superfluid-finance/sdk-core": "^0.8.0",
> ```
> 
> probably not intended (?)

Our rules for semvers of packages is coded in .syncpackrc.yml, this currently requires 'exact' version for prod packages except metadata and ethereum-contracts:

```
semverGroups:
  # Note: rule priority is top (high) to bottom (low)
  - label: Use looser ranges for metadata and ethereum-contracts local packages
    dependencies:
      - "@superfluid-finance/metadata"
      - "@superfluid-finance/ethereum-contracts"
    range: "^"
  - label: Use exact version for prod dependencies
    dependencyTypes: ["prod"]
    range: ""
  - label: Use looser range by default
    range: "^"
```

Lerna has a option `--exact` when managing versions.

But I'd like to review the semverGroups rule before closing the PR, since I find the exceptions "fishy".

**Pull request merged**